lib/beaker/dsl/install_utils/pe_utils.rb

```diff
@@ -512,7 +516,11 @@ def create_agent_specified_arrays(hosts)
           non_agent_only_roles = %w(master database dashboard console frictionless)
           hosts.each do |host|
             if host['roles'].none? {|role| non_agent_only_roles.include?(role) }
-              hosts_agent_only << host
+              if host[:pe_ver] && version_is_less(host[:pe_ver], '4.0')
```

Comment:
I see this check twice - here and again in `roles.rb`. Can this be moved to a method? `is_aio_capable?` or something?